spec/outputs/gcs/path_factory_spec.rb

```diff
@@ -0,0 +1,122 @@
+# encoding: utf-8
+require 'logstash/outputs/gcs/path_factory'
+
+describe LogStash::Outputs::Gcs::PathFactory do
+  describe '#initialize' do
+    it 'includes optional fields if requested' do
+      pf = LogStash::Outputs::Gcs::PathFactory.new(
+          'path/to/directory',
+          'prefix',
+          true,
+          '',
+          true,
+          true,
+          true
+      )
+
+      vars = {
+          prefix: 'prefix',
+          host: 'hostname',
+          date: '2018-01-01',
+          uuid: '00000000-0000-0000-0000-000000000000',
+          partf: '333'
+      }
+
+      expected = 'prefix_hostname_2018-01-01.part333.00000000-0000-0000-0000-000000000000.log.gz'
+      expected = File.join('path/to/directory', expected)
+
+      actual = pf.current_path(vars)
+
+      expect(actual).to eq(expected)
+    end
+
+    it 'excludes optional fields if not requested' do
+      pf = LogStash::Outputs::Gcs::PathFactory.new(
+          'path/to/directory',
+          'prefix',
+          false,
+          '',
+          false,
+          false,
+          false
+      )
+
+      vars = {
+          prefix: 'prefix',
+          host: 'hostname',
+          date: '2018-01-01',
+          uuid: '00000000-0000-0000-0000-000000000000',
+          partf: '333'
+      }
+
+      expected = 'prefix_2018-01-01.log'
+      expected = File.join('path/to/directory', expected)
+
+      actual = pf.current_path(vars)
+
+      expect(actual).to eq(expected)
+    end
+
+    it 'loads a path immediately' do
+      pf = LogStash::Outputs::Gcs::PathFactory.new('', '', false, '', false, false, false)
```

Comment:
It is hard to follow what is going on here. Consider using a builder pattern to initialize the `PathFactory`?
https://medium.com/kkempin/builder-design-pattern-in-ruby-dfa2d557ff1b